presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -1346,6 +1365,7 @@ private void prepareAlterPartition(MetastoreContext metastoreContext, HdfsContex
             }
             partition = Partition.builder(partition)
                     .setCreateTime(oldPartition.get().getCreateTime())
+                    .setLastDataCommitTime(oldPartition.get().getLastDataCommitTime())
```

Comment:
is this correct? shouldn't the lastDataCommitTime be for the new data?

Reply:
Confirmed the DEPA and metastore; the lastDataCommitTime will not be used. Its value of a partition is only updated when the location of the partition is changed.

Reply:
altering the partition does change its location though.

Reply:
> altering the partition does change its location though.

After offline discussion, we know that the lastDataCommitTime field will be ignored by Metastore. Will add the comment about it.